templates/hooks/certificate_authority_activation/sdk_create_post_request.go.tpl

```diff
@@ -0,0 +1,17 @@
+    if err != nil {
+		input := &svcsdk.DescribeCertificateAuthorityInput{}
+		input.CertificateAuthorityArn = desired.ko.Spec.CertificateAuthorityARN
+
+		var describeResp *svcsdk.DescribeCertificateAuthorityOutput
+		describeResp, describeErr := rm.sdkapi.DescribeCertificateAuthorityWithContext(ctx, input)
+		rm.metrics.RecordAPICall("READ_ONE", "DescribeCertificateAuthority", err)
+		if describeErr != nil {
+			return desired, ackrequeue.NeededAfter(describeErr, ackrequeue.DefaultRequeueAfterDuration)
+		}
+
+		if *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusFailed && *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDeleted && *describeResp.CertificateAuthority.Status != svcsdk.CertificateAuthorityStatusDisabled {
+			return desired, ackrequeue.NeededAfter(err, ackrequeue.DefaultRequeueAfterDuration)
+		}
+
+		return nil, err
+	}
```

Comment:
This is duplicate code, can we move this logic into a function in `hooks.go` and reuse it here?

Reply:
This code is used in 2 different resources, would we need to have the function in each resource's `hooks.go` file?